homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,97 @@
+"""This component provides support for Reolink motion events."""
+import asyncio
+import datetime
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+
+from .const import EVENT_DATA_RECEIVED
+from .entity import ReolinkEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DEVICE_CLASS = "motion"
+
+
+@asyncio.coroutine
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the Reolink IP Camera switches."""
+    sensor = MotionSensor(hass, config_entry)
+    async_add_devices([sensor], update_before_add=False)
+
+
+class MotionSensor(ReolinkEntity, BinarySensorEntity):
+    """An implementation of a Reolink IP camera motion sensor."""
+
+    def __init__(self, hass, config):
+        """Initialize a the switch."""
+        ReolinkEntity.__init__(self, hass, config)
+        BinarySensorEntity.__init__(self)
+
+        self._event_state = False
+        self._last_motion = datetime.datetime.now()
+
+    @property
+    def unique_id(self):
+        """Return Unique ID string."""
+        return f"reolink_motion_{self._base.api.mac_address}"
```

Comment:
This is not unique for NVRs, maybe add channel to the mac address.

Reply:
Thanks for the suggestion. Channel is changable from the options menu, so some random ID would probably be more safe.

Reply:
If you could set the channel on initial setup, and add that to your unique ID, your current integration would work with NVR's without having to build the discovery piece.